src/worldfactory.cpp

```diff
@@ -1414,6 +1414,27 @@ void load_world_option( JsonObject &jo )
     }
 }
 
+void load_external_option( JsonObject &jo )
+{
+    auto name = jo.get_string( "name" );
+    auto stype = jo.get_string( "stype" );
+    if( !get_options().has_option( name ) ) {
+        auto sinfo = jo.get_string( "info" );
+        get_options().add_external( name, "world_default", stype, sinfo, sinfo );
+    }
+    if( stype == "float" ) {
+        get_options().get_option( name ).setValue( ( float ) jo.get_float( "value" ) );
```

Comment:
Same here, don't call `get_option` in all branches, call it once and store the returned reference and call `setValue` on that reference.